### ISSUE TITLE
How do I unmap/remap <Tab> in insert mode?

### ISSUE THREAD

**Issue by Brutus11111** [vimwiki/vimwiki#1227]
Prior to submitting a new issue make sure to complete these steps:

- [x] Checkout the `dev` branch and confirm the issue is present there as well.
      The `dev` branch contains fixes that may not have been merged to `master` yet.

not an actual issue i think, just need help

- [x] Post the syntax you are using (default/mediawiki/markdown) **and** your vimwiki settings from your `.vimrc`

```
" vimwiki - Personal Wiki for Vim
" https://github.com/vimwiki/vimwiki
set nocompatible
filetype plugin on
syntax on
" vimwiki with markdown support
let g:vimwiki_ext2syntax = {'.md': 'markdown', '.markdown': 'markdown', '.mdown': 'markdown', '.wiki': 'markdown'}
 

let g:vimwiki_list = [{'path': '~/vimwiki/index.wiki',
                       \ 'syntax': 'markdown', 'ext': '.md'}]

au BufRead,BufNewFile *.wiki set filetype=vimwiki
function! ToggleCalendar()
  execute ":Calendar"
  if exists("g:calendar_open")
    if g:calendar_open == 1
      execute "q"
      unlet g:calendar_open
    else
      g:calendar_open = 1
    end
  else
    let g:calendar_open = 1
  end
endfunction
:autocmd FileType vimwiki map <leader>vc :call ToggleCalendar()
let g:gtd#dir = '~/gtd'
```

- [x] Provide a detailed description of the problem including **steps to reproduce the issue**.

Remap `<tab>`

- [x] Include the output of `:VimwikiShowVersion`.

E492: Not an editor command: VimwikiShowVersion

Ok so here's the problem I'm having and it's probably a super easy fix. This is more of a personal issue, not an issue with the program. 

For most of the things you can easily see in the documentation how to remap something. The one I want to remap (or unmap) is the `<Tab>` one. I'm using coc which uses the tab key by default, which I use a lot when writing LaTeX, but the one in vimwiki is something I never use so I'd like to remap it. with Vimwiki enabled, I can't use the language server it provides, but it does show me the options, I just can't select them. The documentation says:

```
INSERT MODE                                           *vimwiki-    +++table-mappings*
<Tab>                   Go to the next table cell, create a new    
+++row if on the                                                   
  ¦ ¦ ¦ ¦ ¦ ¦ ¦ ¦ ¦ ¦ ¦ last cell.
See |g:vimwiki_table_mappings| to turn them off.
```
Now I'm stuck, I don't know how to remap (or unmap) this feature.

The problem I'm having is that it's mapped like this:

```
i  <Tab>       *@vimwiki#tbl#kbd_tab()  
```
Meanwhile, it's also mapping this:
```
i  <Tab>       * pumvisible() ? "\<C-N>" : <SNR>29_check_back_space() ? "\<Tab>" : coc#ref
```

TLDR: How do I unmap/remap the `<Tab>` in vimwiki?

**Comment by ewal** [comment]
Have you tried disabling `table_mappings`?. I had problems with auto completion in the wiki and this setting solved it for me:
```
let g:vimwiki_key_mappings = { 'table_mappings': 0 }
```

**Comment by Brutus11111** [comment]
> Have you tried disabling `table_mappings`?. I had problems with auto completion in the wiki and this setting solved it for me:
> 
> ```
> let g:vimwiki_key_mappings = { 'table_mappings': 0 }
> ```

Thank you for the comment. I just tried that and it doesn't appear to work. The mapping results are the same.

**Comment by Brutus11111** [comment]
> The above didn't work for me either @Brutus11111
> 
> What I did was removed the bindings for `<TAB>` and `<S-TAB>` from vimwiki/ftplugin/vimwiki.vim
> 
> But I _really_ don't use the Tab key for jumping between links, not sure if you'll want to use this approach.

I commented it out, but it doesn't work for me sadly. 

pack/plugins/start/vimwiki/ftplugin/vimwiki.vim 

```
< "if !hasmapto('<Plug>VimwikiNextLink')
<   "nmap <silent><buffer> <TAB> <Plug>VimwikiNextLink
< "endif
< "nnoremap <silent><script><buffer>
<       "\ <Plug>VimwikiNextLink :VimwikiNextLink<CR>
---
> if !hasmapto('<Plug>VimwikiNextLink')
>   nmap <silent><buffer> <TAB> <Plug>VimwikiNextLink
> endif
> nnoremap <silent><script><buffer>
>       \ <Plug>VimwikiNextLink :VimwikiNextLink<CR>
407,411c407,411
< "if !hasmapto('<Plug>VimwikiPrevLink')
<   "nmap <silent><buffer> <S-TAB> <Plug>VimwikiPrevLink
< "endif
< "nnoremap <silent><script><buffer>
<       "\ <Plug>VimwikiPrevLink :VimwikiPrevLink<CR>
---
> if !hasmapto('<Plug>VimwikiPrevLink')
>   nmap <silent><buffer> <S-TAB> <Plug>VimwikiPrevLink
> endif
> nnoremap <silent><script><buffer>
>       \ <Plug>VimwikiPrevLink :VimwikiPrevLink<CR>

```

**Comment by Brutus11111** [comment]
Issue is fixed by these modifications.

**Comment by KarlWithK** [comment]
This is not really a fix :(

**Comment by krzkrzkrz** [comment]
For anyone still looking. You need to remap `<Plug>VimwikiNextLink` and `<Plug>VimwikiPrevLink` to something else.

For example. Using the Lazy plugin manager:

```
return {
  'vimwiki/vimwiki',
  init = function()
    -- ..
  end,
  config = function()
    vim.g.vimwiki_key_mappings = {
      table_mappings = 0,
    }

    vim.keymap.set('n', '<leader>nl', '<Plug>VimwikiNextLink', { silent = true }) -- For Tab
    vim.keymap.set('n', '<leader>pl', '<Plug>VimwikiPrevLink', { silent = true }) -- For STab
  end
}
```
